### PR TITLE
Add caller information to logging proposals.

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- add caller field to logs
**Related issue(s)**

### CODE REVIEW

File: internal/proposals/logs/improvement-of-log-messages-usability.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ To unify the logs, which will make the debugging process and logs parsing much e
 - **context** - structure of the contextual information, such as operation (for example: `starting workers`), handler/ resolver (for example: `ProvisionRuntime`), controller, resource-namespaced name (for example: `production/application1`), operation ID, instance ID, operation stage, and so on. Users must be able to filter the logs so all the info provided here must be a useful and unique minimal set for every operation. Users must be able to find the needed resource in some store so provide here a name instead of an ID if it's easier to use later.
 - **traceid** - 16-byte numeric value as a base16-encoded string. It'll be passed through a header, so the user can filter all the logs regarding the whole business operation in the whole system.
 - **spanid** - 16-byte numeric value as a base16-encoded string. It'll be randomly generated for each request handling so the user can filter the component logs for a specific operation handling.
+- **caller** - string with information from where the log was created eg.: `"pkg/components/client.go:45"`. It's helpful to quickly track the log.
```
Comment (franpog859) [inline internal/proposals/logs/improvement-of-log-messages-usability.md:42]: You could add it also to the examples

Comment (dbadura) [reply internal/proposals/logs/improvement-of-log-messages-usability.md:42]: Done!

File: internal/proposals/logs/improvement-of-log-messages-usability.md [hunk 2]
```diff
@@ -81,6 +82,7 @@ In the `deployment.yaml` file or other component's container specification, ther
   },
   "traceid": "0354af75138b12921",
   "spanid": "14c902d73a"
```
Comment (franpog859) [inline internal/proposals/logs/improvement-of-log-messages-usability.md:84]:
```suggestion
  "spanid": "14c902d73a",
```

File: internal/proposals/logs/improvement-of-log-messages-usability.md (hunk 1 above)
Comment (franpog859) [inline internal/proposals/logs/improvement-of-log-messages-usability.md:42]:
```suggestion
- **caller** - string with information from where the log was created eg.: `"pkg/components/client.go:45"`. It's helpful to quickly track the log.  
- **traceid** - 16-byte numeric value as a base16-encoded string. It'll be passed through a header, so the user can filter all the logs regarding the whole business operation in the whole system.
- **spanid** - 16-byte numeric value as a base16-encoded string. It'll be randomly generated for each request handling so the user can filter the component logs for a specific operation handling.
```

File: internal/proposals/logs/improvement-of-log-messages-usability.md [hunk 3]
```diff
@@ -81,6 +82,7 @@ In the `deployment.yaml` file or other component's container specification, ther
...
+  "caller": "pkg/components/client.go:45",
```
Comment (dbadura) [inline internal/proposals/logs/improvement-of-log-messages-usability.md:85]:
```suggestion
  "caller": "pkg/components/client.go:45"
```

**Review by franpog859 — APPROVED** [review]

**Review by janmedrek — APPROVED** [review]

File: internal/proposals/logs/improvement-of-log-messages-usability.md [hunk 4]
```diff
@@ -37,6 +37,7 @@ To unify the logs, which will make the debugging process and logs parsing much e
     - message language: English. To preserve unified number/date formatting, go with the `en-us` locale.
     - encoding: `UTF-8`.
 - **context** - structure of the contextual information, such as operation (for example: `starting workers`), handler/ resolver (for example: `ProvisionRuntime`), controller, resource-namespaced name (for example: `production/application1`), operation ID, instance ID, operation stage, and so on. Users must be able to filter the logs so all the info provided here must be a useful and unique minimal set for every operation. Users must be able to find the needed resource in some store so provide here a name instead of an ID if it's easier to use later.
+- **caller** - string with information from where the log was created eg.: `"pkg/components/client.go:45"`. It's helpful to quickly track the log.
```
Comment (majakurcius) [inline internal/proposals/logs/improvement-of-log-messages-usability.md:40]:
```suggestion
- **caller** - string with information where the log was created, where it comes from. For example, `"pkg/components/client.go:45"`. It allows for quick log tracking.  
```
`It's helpful to quickly track the log.` might be read as "Quick log tracking helps.", and what you mean here is quite the opposite. You want to say "Such information helps with quick log tracking, makes quick log tracking easier.", right?

File: internal/proposals/logs/logs-stream-destination-and-library-recomendation.md [hunk 5]
```diff
@@ -51,6 +51,7 @@ Pros:
 - Configurable level of filtering
 - Possibility to set the date format
 - Available log levels: `ERROR`, `INFO`, `FATAL`, `DEBUG`, `PANIC`, `WARN`, `DPANIC` (for development)
+- Support `Caller` field
```
Comment (majakurcius) [inline internal/proposals/logs/logs-stream-destination-and-library-recomendation.md:54]:
```suggestion
- Support for the `Caller` field
```
Or ```- (It) Supports the `Caller` field```

File: internal/proposals/logs/logs-stream-destination-and-library-recomendation.md [hunk 6]
```diff
@@ -68,6 +69,7 @@ Pros:
 - Possibility to set the date format
 - Configurable level of filtering
 - Available log levels: `ERROR`, `INFO`, `FATAL`, `DEBUG`, `PANIC`, `WARN`, `TRACE`, no level
+- Support `Caller` field
```
Comment (majakurcius) [inline internal/proposals/logs/logs-stream-destination-and-library-recomendation.md:72]:
```suggestion
- Support for the `Caller` field
```

File: internal/proposals/logs/logs-stream-destination-and-library-recomendation.md [hunk 7]
```diff
@@ -84,6 +86,7 @@ Pros:
 
 Cons:
 
+- Lack of `Caller` field
```
Comment (majakurcius) [inline internal/proposals/logs/logs-stream-destination-and-library-recomendation.md:89]:
```suggestion
- No `Caller` field
```
There's a subtle difference between `lack of` and `no`. `Lack of` means that there's not enough (of) something, or something isn't available (like, lack of money preventing you from doing sth). 
What you mean here, I believe, is just stating the fact that this library doesn't have this field.

File: internal/proposals/logs/improvement-of-log-messages-usability.md (hunk 4 above)
Comment (dbadura) [reply internal/proposals/logs/improvement-of-log-messages-usability.md:40]: Yes

File: internal/proposals/logs/logs-stream-destination-and-library-recomendation.md (hunk 5 above)
Comment (dbadura) [reply internal/proposals/logs/logs-stream-destination-and-library-recomendation.md:54]:
```suggestion
- Supports the `Caller` field
```
Looks better for me

File: internal/proposals/logs/improvement-of-log-messages-usability.md (hunk 4 above)
Comment (dbadura) [reply internal/proposals/logs/improvement-of-log-messages-usability.md:40]:
```suggestion
- **caller** - string with information where in the code the log was created. For example: `"pkg/components/client.go:45"`.
```

File: internal/proposals/logs/poc-streams-and-libraries/apex/apex.go [hunk 8]
```diff
@@ -8,11 +8,10 @@ import (
 
 func main() {
 	log.SetHandler(json.New(os.Stderr))
-
 	testApexLogger(log.WithField("request_id", "random_string"))
 }
 
 func testApexLogger(logEntry *log.Entry) {
 	logEntry.WithField("context", "a").Infof("just normal log with msg: %s", "Hello From Apex")
```
Comment (majakurcius) [inline internal/proposals/logs/poc-streams-and-libraries/apex/apex.go:15]:
```suggestion
	logEntry.WithField("context", "a").Infof("just a normal log entry with the msg: %s", "Hello From Apex")
```

File: internal/proposals/logs/poc-streams-and-libraries/apex/apex.go [hunk 9]
```diff
@@ -8,11 +8,10 @@ import (
...
-	logEntry.Errorf("Error msg: %s", "some error occured")
+	logEntry.Errorf("Error msg: %s", "some error occurred")
```
Comment (majakurcius) [inline internal/proposals/logs/poc-streams-and-libraries/apex/apex.go:16]:
```suggestion
	logEntry.Errorf("Error msg: %s", "an error occurred")
```

**Review by majakurcius — COMMENTED** [review]
And part 2

**Review by majakurcius — APPROVED** [review]
image

**Pull request merged**